20_classes_lab/03.py

```diff
@@ -1,13 +1,33 @@
-"""
-Following code assumes a class named Widget that depends on
-other widgets
-Widget ctor takes a name, and function 
-add_dependency(depends)
-marks widget as depends on all widgets in "depends" list
-
-When building a widget, you need to first build all the widgets it
-depends on.
-
-Implement widget so the following code works
-"""
-
+class Widget (object):
+    def __init__(self, val = ""):
+        self.name = val
+        self.depends = []
+
+    def add_dependency(self, *dependencies):
+        for dependency in dependencies:
+            self.depends.append(dependency)
+
+    def build(self):
+        if self.depends != []:
+            for widget in self.depends:
+                print widget.name

```

Comment:
Not good enough: You should also print all widget's dependencies if they were not already printed